packages/@mantine/charts/src/BarChart/BarChart.tsx

```diff
@@ -130,10 +130,23 @@ export function BarLabel({
   parentViewBox,
   ...others
 }: Record<string, any>) {
+
+  function labelCoordinates(): { dx: number, dy: number } {
+    let coord: { dx: number, dy: number } = { dx: 0, dy: 0 }
+    if (others.props.h < 300) {
+      coord = { ...coord, dx: Math.ceil(others?.width + 10) }
+    } else {
+      coord.dx = Math.ceil(others?.width - others?.viewBox?.x / 2)
+    }
+    coord = { ...coord, dy: Math.ceil(others.props.h / 15) }
+    return coord;
+  }
```

Comment:
IMO: I think this way is much simple, WDYT? I prefer not using `let` in this case.

```suggestion
function labelCoordinates(): { dx: number, dy: number } {
  return {
    dx: others.props.h < 300
      ? Math.ceil(others?.width + 10)
      : Math.ceil(others?.width - others?.viewBox?.x / 2),
    dy: Math.ceil(others.props.h / 15)
  };
}
```

Reply:
That looks good too.
I tried not to use ternary operators just in case we need to add more conditions in future.

Reply:
Since the code is not complicated yet, I think we should consider it only when additional conditions arise!